jwst/ramp_fitting/utils.py

```diff
@@ -292,19 +289,20 @@ def output_optional(self, effintim):
         self.weights[1. / self.weights > 0.4 * LARGE_VARIANCE] = 0.
         warnings.resetwarnings()
 
-        rfo_model = \
-            datamodels.RampFitOutputModel(
-                slope=self.slope_seg.astype(np.float32) / effintim,
-                sigslope=self.sigslope_seg.astype(np.float32),
-                var_poisson=self.var_p_seg.astype(np.float32),
-                var_rnoise=self.var_r_seg.astype(np.float32),
-                yint=self.yint_seg.astype(np.float32),
-                sigyint=self.sigyint_seg.astype(np.float32),
-                pedestal=self.ped_int.astype(np.float32),
-                weights=self.weights.astype(np.float32),
-                crmag=self.cr_mag_seg)
+        slope = self.slope_seg.astype(np.float32) / effintim
+        sigslope = self.sigslope_seg.astype(np.float32)
+        var_poisson = self.var_p_seg.astype(np.float32)
+        var_rnoise = self.var_r_seg.astype(np.float32)
+        yint = self.yint_seg.astype(np.float32)
+        sigyint = self.sigyint_seg.astype(np.float32)
+        pedestal = self.ped_int.astype(np.float32)
+        weights = self.weights.astype(np.float32)
```

Comment:
Specifying `astype()` here will guarantee a copy happens here instead of later when they get assigned to the datamodel, where the `dtype` is actually enforced.  It would probably be best to do the `dtype` enforcement when assigned to the datamodel?  This is basically redundant to the `RampFitOutputModel` schema:

https://github.com/spacetelescope/jwst/blob/master/jwst/datamodels/schemas/rampfitoutput.schema.yaml

Reply:
I changed this to do the slope computation, then return a tuple of each array of interest in the `OptRes` class, instead of using `astype` to create a copy here.